js/index.js

```diff
@@ -37,6 +37,66 @@ const siteContent = {
   },
 };
 
-// Example: Update the img src for the logo
-let logo = document.getElementById("logo-img");
-logo.setAttribute('src', siteContent["nav"]["img-src"])
+//TASK 2: UPDATE IMAGE PATHS
+let imgSetter = (id, src) => document.getElementById(id).setAttribute('src', src);
+imgSetter('logo-img', siteContent.nav["img-src"]);
+imgSetter('cta-img', siteContent.cta["img-src"]);
+imgSetter('middle-img', siteContent["main-content"]["middle-img-src"]);
+
+//TASK 3: UPDATE TEXT CONTENT
+//NAV
+//filter out text tags from nav then add text from siteContent to each remaining (anchor) tag
+//steal filter method from array class and apply it to childNodes
+[].filter.call(document.getElementsByTagName('nav')[0].childNodes, node => node.nodeType === Node.ELEMENT_NODE)
+  .forEach((node, i) => {
+    node.textContent = siteContent.nav[`nav-item-${(i+1)}`];
+    node.setAttribute('style', 'color: green;'); //TASK 4
+});
+
+//CTA
+document.getElementsByClassName("cta-text")[0].childNodes
+.forEach(node => node.textContent = siteContent.cta[node.nodeName.toLowerCase()]);
+
+//MAIN CONTENT
+//grab each text-content inside of main-content
+[].forEach.call(document.getElementsByClassName('main-content')[0].getElementsByClassName('text-content'), txtContent => {
+//filter out all the text tags and grab all the h4's and p's
+[].filter.call(txtContent.childNodes, node => node.nodeType === Node.ELEMENT_NODE)
+  .forEach(node => {
+    //loop through main-content object, remove img if found, add text to node if not #jankyButWorks
+    for(let key in siteContent['main-content']) { //for loop because how else do I get the first prop in an object?
```

Comment:
I would like to see this cleaned up, some times splitting code into multiple sections isn't a bad thing as you also have to think about readability with other team members that might come behind you. I wouldn't advise also using an empty array and using call as that is better fit to a binding method.